hypex/operators/operators.py

```diff
@@ -75,39 +101,172 @@ def _execute_inner_function(
+        data_space: ExperimentDataEnum = ExperimentDataEnum.groups,
+        executor_name: str = "FaissNearestNeighbors",
+        key: Any = "",
+    ):
+        super().__init__(
+            grouping_role=grouping_role, target_roles=target_roles, space=space, key=key
+        )
+        self.data_space = data_space
+        self.executor_name = executor_name
+
+    def execute(self, data: ExperimentData) -> ExperimentData:
+        group_field, target_fields = self._get_fields(data)
+        t_data = deepcopy(data.ds)
+        distances_keys = data.get_ids(self.executor_name, self.data_space)
+        if (
+            len(distances_keys[self.executor_name]["groups"]) > 0
+        ):  # добавить поиск в analysis_tables
+            matched_data_key = list(
+                data.groups[distances_keys[self.executor_name]["groups"][0]].keys()
+            )[0]
+            matched_data = data.groups[distances_keys[self.executor_name]["groups"][0]][
+                matched_data_key
+            ]
+            target_fields += matched_data.search_columns(
+                self.target_roles
+            )  # нужно проверять, что таргетов не 2
+        else:
+            raise NotFoundInExperimentDataError(self.executor_name)
+        self.key = str(
+            target_fields[0] if len(target_fields) == 1 else (target_fields or "")
+        )
+        if (
+            not target_fields and data.ds.tmp_roles
+        ):  # если колонка не подходит для теста, то тагет будет пустой, но если есть темп роли, то это нормальное поведение
+            return data
+
+        compare_result = self.calc(
+            data=t_data.append(matched_data, axis=1),
+            group_field=group_field,
+            target_fields=target_fields,
+            features_fields=t_data.search_columns(
+                FeatureRole(), search_types=[int, float]
+            )
+            + matched_data.search_columns(FeatureRole(), search_types=[int, float]),
+        )
+        return self._set_value(data, compare_result)
+
+    @classmethod
+    def _execute_inner_function(
+        cls,
+        grouping_data,
+        target_fields: Optional[List[str]] = None,
+        features_fields: Optional[List[str]] = None,
+        **kwargs
+    ) -> Dict:
+        return cls._inner_function(
+            grouping_data[0][1],
+            test_data=grouping_data[1][1],
+            target_fields=target_fields,
+            features_fields=features_fields,
+            **kwargs
+        )
+
     @classmethod
     def _inner_function(
-        cls, data: Dataset, test_data: Optional[Dataset] = None, **kwargs
-    ) -> Any:
-        raise NotImplementedError
+        cls,
+        data: Dataset,
+        test_data: Optional[Dataset] = None,
+        target_fields: Optional[List[str]] = None,
+        features_fields: Optional[List[str]] = None,
+        **kwargs
+    ) -> Dict:
+        if target_fields is None or features_fields is None or test_data is None:
```

Comment:
Why do we have these variables as optional arguments if we raise error if they are None?